### PR TITLE
Feature/auto populate

### PR DESCRIPTION
**Pull request by jacquesporveau**
https://www.pivotaltracker.com/story/show/141823283

### CODE REVIEW

File: app/controllers/presentations_controller.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@ def show
 
   def new
     @presentation = Presentation.new
+    @event = Event.find(params[:event_id]) if params[:event_id]
```
Comment (graygilmore) [inline app/controllers/presentations_controller.rb]: I think you can remove the conditional here. If `params[:id]` is `nil` then it won't be able to find an Event and `@event` ends up being `nil`, too.

* pretty sure

**Review by graygilmore — CHANGES_REQUESTED** [review]

File: app/controllers/presentations_controller.rb (hunk 1 above)
Comment (jacquesporveau) [reply app/controllers/presentations_controller.rb]: Actually it 500's. If we did a `.find_by(id: params[:event_id])` that would work but I don't want to set an instance variable to `nil` if I don't have to.

**Review by graygilmore — CHANGES_REQUESTED** [review]
Looks like this is missing from the home page sign up link.

**Comment by jacquesporveau** [comment]
Mmmmmm yep alright that was sloppy. I've gone through and done my best to nail all the spots that this new link should be in. Worth noting that we could probably use more sign up links throughout the app. I added a bit about that to the QA audit issue.

File: app/controllers/presentations_controller.rb (hunk 1 above)
Comment (qr8r) [reply app/controllers/presentations_controller.rb]: In the next commit though you call `@event&.id` anyway though. when you call an instance variable it is created with the initial value of nil. 

i'd either to the `find_by` or something like `@event = Event.find_by(id: params[:event_id]) || Event.new` 
that way you're always operating on an event object. 

Id also consider always building the presentation off of the event object `@event.presentations.build`

Comment (jacquesporveau) [reply app/controllers/presentations_controller.rb]: Oh that`.build` is cool. I didn't know you can do that / or it worked that way.

Comment (qr8r) [reply app/controllers/presentations_controller.rb]: http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html

Comment (jacquesporveau) [reply app/controllers/presentations_controller.rb]: This comment is showing up as not updated but if you view the diff it actually has been changed.

**Review by qr8r — APPROVED** [review]

**Review by graygilmore — APPROVED** [review]

**Pull request merged**